src/ASM/ASMs2DLag.C

```diff
@@ -773,3 +773,45 @@ bool ASMs2DLag::write(std::ostream& os, int) const
 {
   return this->writeLagBasis(os, "quad");
 }
+
+
+int ASMs2DLag::findElement(double u, double v,
+                           double* xi, double* eta) const
+{
+  double du = 1.0 / (nx-1);
+  double dv = 1.0 / (ny-1);
+
+  int elmx = std::min(nx-2.0, floor(u / du));
```

Comment:
Wouldn't `std::min(nx-2.0, floor(u*(nx-1)))` yield the same? Don't see why you need the temporary du here.
Or alternatively (since elmx is and int): `std::min(nx-2, (int)floor(u*(nx-1)))`.

Reply:
Not needed, I think it increases readability with the temporary, but I've changed to how you want it.